diff --git a/testbox/system/coverage/CoverageService.cfc b/testbox/system/coverage/CoverageService.cfc
--- a/testbox/system/coverage/CoverageService.cfc
+++ b/testbox/system/coverage/CoverageService.cfc
@@ -175 +175 @@
-		if ( isNull( opts.coverageTresholds ) ) {
+		if ( !structKeyExists( opts, "coverageTresholds" ) || isNull( opts.coverageTresholds ) ) {
@@ -178 +178 @@
-		if ( isNull( opts.coverageTresholds.good ) ) {
+		if ( !structKeyExists( opts.coverageTresholds, "good" ) || isNull( opts.coverageTresholds.good ) ) {
@@ -181 +181 @@
-		if ( isNull( opts.coverageTresholds.bad ) ) {
+		if ( !structKeyExists( opts.coverageTresholds, "bad" ) || isNull( opts.coverageTresholds.bad ) ) {
