diff --git a/testbox/system/coverage/CoverageService.cfc b/testbox/system/coverage/CoverageService.cfc
--- a/testbox/system/coverage/CoverageService.cfc
+++ b/testbox/system/coverage/CoverageService.cfc
@@ -172,8 +172,8 @@ component accessors="true" {
 			}
 		}
 
-		if ( isNull( opts.coverageTresholds ) ) {
+		if ( !structKeyExists( opts, "coverageTresholds" ) || isNull( opts.coverageTresholds ) ) {
 			opts.coverageTresholds = {};
 		}
 		if ( isNull( opts.coverageTresholds.good ) ) {
 			opts.coverageTresholds.good = 85;
